### PR TITLE
doc/rados: edit pools.rst (2 of x)

### PR DESCRIPTION
**Pull request by zdover23**
Edit doc/operations/rados/pools.rst.

There remains confusion in this part of the document regarding pg_num and pgp_num. pg_num and pgp_num are not explained with sufficient clarity. A future commit will clear up this confusion. There is also some potential confusion between on the one hand the strings "pg-num" and "pgp-num" and on the other hand "pg_num" and "pgp_num". The strings with the hyphens are used in dummy commands, and the strings with the underscores are used as key names. I think it possible that this could confuse a reader, but I am open to discussion on the matter.

https://tracker.ceph.com/issues/58485

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by zdover23** [comment]
jenkins retest this please

File: doc/rados/operations/pools.rst [hunk 1]
```diff
@@ -178,14 +178,14 @@ following:
...
-Pools need to be associated with an application before use. Pools that will be
-used with CephFS or pools that are automatically created by RGW are
-automatically associated. Pools that are intended for use with RBD should be
-initialized using the ``rbd`` tool (see `Block Device Commands`_ for more
-information).
+Pools need to be associated with an application before they can be used. Pools
+that are intended for use with CephFS and pools that are automatically created
+by RGW are associated automatically. Pools that are intended for use with RBD
+should be initialized with the ``rbd`` tool (see `Block Device Commands`_ for
+more information).
```
Comment (anthonyeleven) [inline doc/rados/operations/pools.rst:185]: created automatically?  Remember the "Boldly go" rule ;)

File: doc/rados/operations/pools.rst [hunk 2]
```diff
@@ -194,11 +194,11 @@ a pool.:
...
 
-Set Pool Quotas
-===============
+Setting Pool Quotas
+===================
 
-You can set pool quotas for the maximum number of bytes and/or the maximum
-number of objects per pool:
+To set pool quotas for the maximum number of bytes and/or the maximum
+number of objects per pool, run the following command:
```
Comment (anthonyeleven) [inline doc/rados/operations/pools.rst:201]: RADOS objects

File: doc/rados/operations/pools.rst [hunk 3]
```diff
@@ -213,103 +213,101 @@ For example:
...
-
-See `Monitor Configuration`_ for more information.
+For more information, see `Monitor Configuration`_.
 
 .. _Monitor Configuration: ../../configuration/mon-config-ref
 
-If you created your own rules for a pool you created, you should consider
-removing them when you no longer need your pool:
+If you created custom rules for a pool that you created but the pool is no longer needed,
+consider deleting those rules.
```
Comment (anthonyeleven) [inline doc/rados/operations/pools.rst:233]: Suggest
>If there are custom rules for a pool that is no longer needed,

File: doc/rados/operations/pools.rst [hunk 4]
```diff
@@ -213,103 +213,101 @@ For example:
...
 
 .. prompt:: bash $
 
    ceph osd pool get {pool-name} crush_rule
 
-If the rule was "123", for example, you can check the other pools like so:
+For example, if the custom rule is "123", you can check all other pools to see whether they use the rule by running the following command:
```
Comment (anthonyeleven) [inline doc/rados/operations/pools.rst:239]: s/you can //
s/other//

File: doc/rados/operations/pools.rst [hunk 5]
```diff
@@ -213,103 +213,101 @@ For example:
...
 
 .. prompt:: bash $
 
-	ceph osd dump | grep "^pool" | grep "crush_rule 123"
+    ceph osd dump | grep "^pool" | grep "crush_rule 123"
 
-If no other pools use that custom rule, then it's safe to delete that
+If no other pools use this custom rule, then it is safe to delete the
```
Comment (anthonyeleven) [inline doc/rados/operations/pools.rst:245]: s/other//

File: doc/rados/operations/pools.rst [hunk 6]
```diff
@@ -213,103 +213,101 @@ For example:
...
 rule from the cluster.
 
-If you created users with permissions strictly for a pool that no longer
-exists, you should consider deleting those users too:
-
+Similarly, if you created users with permissions restricted to a pool that no longer
```
Comment (anthonyeleven) [inline doc/rados/operations/pools.rst:248]: >if there are users

They may be inherited or created by someone else.

File: doc/rados/operations/pools.rst [hunk 7]
```diff
@@ -213,103 +213,101 @@ For example:
...
 .. prompt:: bash $
 
    ceph osd pool rename {current-pool-name} {new-pool-name}
 
-If you rename a pool and you have per-pool capabilities for an authenticated
-user, you must update the user's capabilities (i.e., caps) with the new pool
-name.
+If you rename a pool for which an authenticated user has per-pool capabilities,
+you must update the user's capabilities ("caps") to refer to the new pool name.
+
```
Comment (anthonyeleven) [inline doc/rados/operations/pools.rst:268]: 😮

File: doc/rados/operations/pools.rst [hunk 8]
```diff
@@ -213,103 +213,101 @@ For example:
...
 .. _setpoolvalues:
 
+Setting Pool Values
+===================
 
-Set Pool Values
-===============
-
-To set a value to a pool, execute the following:
+To set pool values for keys, run a command of the following form:
```
Comment (anthonyeleven) [inline doc/rados/operations/pools.rst:310]: To assign values to a pool's configuration keys, run

File: doc/rados/operations/pools.rst [hunk 9]
```diff
@@ -321,42 +319,47 @@ You may set values for the following keys:
...
+   - :confval:`bluestore_compression_min_blob_size_ssd`
 
    :Type: Unsigned Integer
 
 .. describe:: compression_max_blob_size
 
-   Chunks larger than this are broken into smaller blobs sizing
-   ``compression_max_blob_size`` before being compressed.
+   Sets the maximum size for chunks: that is, chunks larger than this are
+   broken into smaller blobs of this size before compression is carried out.
```
Comment (anthonyeleven) [inline doc/rados/operations/pools.rst:353]: >broken into blobs of at most this size

s/carried out/performed/

File: doc/rados/operations/pools.rst [hunk 10]
```diff
@@ -378,45 +381,41 @@ You may set values for the following keys:
 
 .. describe:: pg_num
 
-   The effective number of placement groups to use when calculating
+   Sets the effective number of PGs to use when calculating
    data placement.
 
    :Type: Integer
-   :Valid Range: Superior to ``pg_num`` current value.
+   :Valid Range: Greater than the current value of ``pg_num``.
```
Comment (anthonyeleven) [inline doc/rados/operations/pools.rst:388]: How has this been this wrong so long?

>The number of PGs to belong to this pool.

:Valid Range: 0 to `mon_max_pool_pg_num`.  If set to 0, the value of `osd_pool_default_pg_num` will be used.

https://github.com/ceph/ceph/blob/b554518133a6fa527cc6b02d7fcf4fb7410e1f0d/src/mon/OSDMonitor.cc#L7958

File: doc/rados/operations/pools.rst [hunk 11]
```diff
@@ -378,45 +381,41 @@ You may set values for the following keys:
...
 
 .. describe:: pgp_num
 
-   The effective number of placement groups for placement to use
-   when calculating data placement.
+   Sets the effective number of PGs to use when calculating data placement.
 
    :Type: Integer
-   :Valid Range: Equal to or less than ``pg_num``.
+   :Valid Range: Less than or equal to the current value of ``pg_num``.
```
Comment (anthonyeleven) [inline doc/rados/operations/pools.rst:397]: Between 1 and the current value of

File: doc/rados/operations/pools.rst [hunk 12]
```diff
@@ -378,45 +381,41 @@ You may set values for the following keys:
...
 
 .. _crush_rule:
 
 .. describe:: crush_rule
 
-   The rule to use for mapping object placement in the cluster.
+   Sets the rule that Ceph uses to map object placement in the cluster.
```
Comment (anthonyeleven) [inline doc/rados/operations/pools.rst:403]: The CRUSH rule that Ceph uses to determine RADOS object placement within the pool.

File: doc/rados/operations/pools.rst [hunk 13]
```diff
@@ -378,45 +381,41 @@ You may set values for the following keys:
...
 
 .. describe:: allow_ec_overwrites
 
-
-   Whether writes to an erasure coded pool can update part
-   of an object, so cephfs and rbd can use it. See
-   `Erasure Coding with Overwrites`_ for more details.
+   Determines whether writes to an erasure-coded pool are allowed
+   to update part of an object so that CephFS and RBS can use the object.
+   For more details, see `Erasure Coding with Overwrites`_.
```
Comment (anthonyeleven) [inline doc/rados/operations/pools.rst:413]: Determines whether writes to an erasure-coded pool may update only part of a RADOS object.  This allows CephFS and RBD to use the pool for data.

File: doc/rados/operations/pools.rst [hunk 14]
```diff
@@ -862,8 +861,8 @@ You may get values for the following keys:
 :Type: Integer
 
 
-Set the Number of Object Replicas
-=================================
+Setting the Number of Object Replicas
```
Comment (anthonyeleven) [inline doc/rados/operations/pools.rst:864]: RADOS Object

**Review by anthonyeleven — APPROVED** [review]
Various changes, non-blocking for your convenience, but several of them address factual errors or egregious speling.

Re `pg_num` and `pgp_num`, I suggest not sweating them.  We no doubt cover that elsewhere in the docs.  Moreover, since Nautilus and Octopus / Pacific, operator awareness is not nearly as important.

Nautilus brought automagic stepping of `pgp_num` to match `pg_num`, something we used to have to manage manually.  In ... Pacific, maybe Octopus, we gained the ability to merge PGs, i.e. to decrease `pg_num` for a given pool.  Prior to this one could only increase the value, so as when naming one's first pet, careful up-front thought needed to be given.

File: doc/rados/operations/pools.rst (hunk 1 above)
Comment (zdover23) [reply doc/rados/operations/pools.rst:185]: This isn't a split infinitive, but I share your zeal for keeping the parts of a complex verb together. So: accepted.

File: doc/rados/operations/pools.rst (hunk 2 above)
Comment (zdover23) [reply doc/rados/operations/pools.rst:201]: Accepted.

File: doc/rados/operations/pools.rst (hunk 3 above)
Comment (zdover23) [reply doc/rados/operations/pools.rst:233]: That is better. Accepted.

File: doc/rados/operations/pools.rst (hunk 4 above)
Comment (zdover23) [reply doc/rados/operations/pools.rst:239]: Well-spotted. Accepted.

File: doc/rados/operations/pools.rst (hunk 5 above)
Comment (zdover23) [reply doc/rados/operations/pools.rst:245]: Accepted.

File: doc/rados/operations/pools.rst (hunk 8 above)
Comment (zdover23) [reply doc/rados/operations/pools.rst:310]: Accepted.

File: doc/rados/operations/pools.rst (hunk 9 above)
Comment (zdover23) [reply doc/rados/operations/pools.rst:353]: Accepted.

File: doc/rados/operations/pools.rst (hunk 10 above)
Comment (zdover23) [reply doc/rados/operations/pools.rst:388]: Accepted. Good research.

And how could it have been wrong for so long? There are so few of us, and so many lines to check.

File: doc/rados/operations/pools.rst (hunk 11 above)
Comment (zdover23) [reply doc/rados/operations/pools.rst:397]: Accepted.

File: doc/rados/operations/pools.rst (hunk 12 above)
Comment (zdover23) [reply doc/rados/operations/pools.rst:403]: Accepted.

File: doc/rados/operations/pools.rst (hunk 13 above)
Comment (zdover23) [reply doc/rados/operations/pools.rst:413]: I'm going to accept this, but I think that the phrase "to use the pool for data" is vague. This is a danger of using for-phrases. So: what exactly does it mean to use a pool for data? (I'm probably going to merge this as it is, but I will have no compunction to raise a future pull request to make this sentence better.)

File: doc/rados/operations/pools.rst (hunk 14 above)
Comment (zdover23) [reply doc/rados/operations/pools.rst:864]: Accepted.

File: doc/rados/operations/pools.rst (hunk 13 above)
Comment (anthonyeleven) [reply doc/rados/operations/pools.rst:413]: As opposed to metadata.  I almost wrote “user data” or “payload data”

File: doc/rados/operations/pools.rst (hunk 10 above)
Comment (anthonyeleven) [reply doc/rados/operations/pools.rst]: I'm embarrassed that *I* didn't notice this before as an operator.

File: doc/rados/operations/pools.rst (hunk 13 above)
Comment (anthonyeleven) [reply doc/rados/operations/pools.rst]: @zdover still need to s/RBS/RBD/

Comment (zdover23) [reply doc/rados/operations/pools.rst]: Oh, good Lord. You are correct.

Comment (zdover23) [reply doc/rados/operations/pools.rst]: Good looking out.

File: doc/rados/operations/pools.rst [hunk 15]
```diff
@@ -320,103 +322,81 @@ You may set values for the following keys:
...
 
 .. describe:: allow_ec_overwrites
-
-
-   Whether writes to an erasure coded pool can update part
-   of an object, so cephfs and rbd can use it. See
-   `Erasure Coding with Overwrites`_ for more details.
-
+   
+   :Description: Determines whether writes to an erasure-coded pool are allowed to update only part of a RADOS object. This allows CephFS and RBD to use the pool for data (as opposed to metadata). For more details, see `Erasure Coding with Overwrites`_.
```
Comment (anthonyeleven) [inline doc/rados/operations/pools.rst:395]: My "as opposed to" comment was worded for discussion.  For inclusion here, I suggest

>This allows CephFS and RBD to use an EC pool for user data (but not for metadata).

File: doc/rados/operations/pools.rst (hunk 13 above)
Comment (anthonyeleven) [reply doc/rados/operations/pools.rst]: See below for refinement

**Comment by zdover23** [comment]
Note to the near-future: It took three minutes for my squashed force push to be registered on this page.

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/51939 - Reef backport
https://github.com/ceph/ceph/pull/51940 - Quincy backport

**Pull request merged**